snuba/datasets/plans/query_plan.py

```diff
@@ -69,6 +69,8 @@ class ClickhouseQueryPlan(QueryPlan[Query]):
     are executed by the execution strategy at every DB Query.
     """
 
+    query: Query
+    execution_strategy: QueryPlanExecutionStrategy[Query]
```

Comment:
Wait, the previous version did not work because of a mypy bug. I think we should mark it as ignore for as long as it gets fixed. I am not sure overriding the type would actually fix it.